packages/react/src/utils/createBaseUIEvent.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import type { BaseUIEvent } from './types';
+import { makeEventPreventable } from '../merge-props/mergeProps';
+import { NOOP } from './noop';
+
+/**
+ * Creates a SyntheticEvent that satisfies the `BaseUIEvent` contract.
+ */
+export function createBaseUIEvent<E extends Event = Event>(
+  eventParam?: React.SyntheticEvent<Element, E> | E,
+): BaseUIEvent<React.SyntheticEvent<Element, E>> {
+  if (eventParam && typeof eventParam === 'object' && 'persist' in eventParam) {
```

Comment:
In general, runtime type detection is a code smell. There seems to be two independent cases, which could be split into something like:
 - `BaseUIEvent.fromNative(nativeEvent)`
 - `BaseUIEvent.fromSynthetic(event)`

Reply:
Adjusted